src/ride/track.c

```diff
@@ -4921,15 +4922,493 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+
+		// 6cdd92
+		mapElement->properties.track.maze_entry &= (1 << edx);
+		return loc_6CDDA2(x, y, rideIndex, mapElement);
+	}
+
+
+	// 0x6cdb53
+	if (direction == 4) {
+		return loc_6CDDA2(x, y, rideIndex, mapElement);
+	}
+
+	// 0x6cdb69
+
+	uint16 ax = x - TileDirectionDelta[direction].x / 2;
+	uint16 cx = x - TileDirectionDelta[direction].y / 2;
+
+	bl = (((mode << 8) + rideIndex) >> 3) & 0xFF;
+	// esi != mapElement
+	uint16 si = cx;
+
+	si = floor2(si, 32);
+	uint16 tmp_ax = floor2(ax, 32);
+
+	uint32 esi = ((si << 8) | tmp_ax) >> 3;
+
+	mapElement = gMapElementTilePointers[esi];
+
+	// eax = incremented X;
+	map_invalidate_tile_full(floor2(ax, 32), floor2(cx, 32));
+
+	// eax = ax;
+	// ecx = cx;
+	// ebx = [bl: bl, bh: rideIndex];
+
+	do {
+		if (map_element_get_type(mapElement) != MAP_ELEMENT_TYPE_TRACK) {
+			continue;
+		}
+
+		if (mapElement->base_height != bl) {
+			continue;
+		}
+
+		if (mapElement->properties.track.ride_index != rideIndex) {
+			continue;
+		}
+
+		if (mapElement->properties.track.type != 0x65) {
+			continue;
+		}
+
+		return loc_6CDBE9(x, y, ax, cx, z, rideIndex, mapElement);
+		// loc_6CDBDB:
+	} while (map_element_is_last_for_tile(mapElement++));
+
+	return loc_6CDDA2(x, y, rideIndex, mapElement);
+}
+
+
+money32 do_game_command_set_maze_track(uint16 x, uint8 flags, uint8 direction, uint16 y, uint8 rideIndex, uint8 mode, uint16 unknown, uint16 z) {
+	uint16 temp_dx;
+
+	// 0x006CD8CE: game_command_set_maze_track
+	RCT2_GLOBAL(RCT2_ADDRESS_NEXT_EXPENDITURE_TYPE, uint8) = 0;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_X, uint16) = x + 8;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_Y, uint16) = y + 8;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_Z, uint16) = z;
+
+	RCT2_GLOBAL(0xF4413E, money32) = 0;
+
+	if (!sub_68B044()) {
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	if (!(flags & GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED) && RCT2_GLOBAL(RCT2_ADDRESS_GAME_PAUSED, uint8) != 0) {
+		// loc_6CDE2D:
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = STR_CONSTRUCTION_NOT_POSSIBLE_WHILE_GAME_IS_PAUSED;
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	// loc_6CD920:
+	if ((z & 0xF) != 0) {
+		// loc_6CDE38:
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = 954;
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	if ((flags & GAME_COMMAND_FLAG_APPLY) != 0) {
+		if ((flags & GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED) == 0) {
+			footpath_remove_litter(x, y, z);
+			map_remove_walls_at((uint16) floor2(x, 32), (uint32) floor2(y, 32), z, z + 0x20);
+		}
+	}
+
+	// loc_6CD954:
+	if (map_is_location_owned(floor2(x, 32), floor2(y, 32), z) == 0) {
+		// loc_6CDE51
+		return MONEY32_UNDEFINED;
+	}
+
+	rct_map_element *mapElement = map_get_surface_element_at(x / 32, y / 32);
+	if (mapElement == NULL) {
+		// TODO: set error text?
+		return MONEY32_UNDEFINED;
+	}
+
+	// loc_6CD99A:
+	temp_dx = z;
+	temp_dx += 0x20;
+	temp_dx = temp_dx >> 3;
+	sint8 temp_dl = (uint8) (temp_dx & 0xFF);
```

Comment:
`sint8 supportHeight = ((z + 32) >> 3) - mapElement->base_height;`

Always try to avoid calling a variable by the register. In this case its very easy to work out what this is for as the error text tells you.